base/tuple.jl

```diff
@@ -73,14 +73,15 @@ keys(@nospecialize t::Tuple) = OneTo(length(t))
 prevind(@nospecialize(t::Tuple), i::Integer) = Int(i)-1
 nextind(@nospecialize(t::Tuple), i::Integer) = Int(i)+1
 
-function keys(t::Tuple, t2::Tuple...)
-    @inline
-    OneTo(_maxlength(t, t2...))
+@noinline function throw_eachindex_mismatch_indices(::Tuple{}, inds...)
```

Comment:
what is the purpose of the first `::Tuple{}` parameter?

Reply:
I think it resolves dispatch:
```
julia> methods(Base.throw_eachindex_mismatch_indices)
# 2 methods for generic function "throw_eachindex_mismatch_indices" from Base:
 [1] throw_eachindex_mismatch_indices(::IndexCartesian, inds...)
     @ abstractarray.jl:324
 [2] throw_eachindex_mismatch_indices(::IndexLinear, inds...)
     @ abstractarray.jl:321
```
It's not obvious to me that one couldn't use the `IndexLinear` dispatch but this seems fine.